ibis/expr/api.py

```diff
@@ -1016,6 +1016,25 @@ def log(arg, base=None):
     return op.to_expr()
 
 
+def clip(arg, lower=None, upper=None):
+    """
+    Trim values at input threshold(s).
+
+    Parameters
+    ----------
+    lower : float
+    upper : float
+
+    Returns
+    -------
+    clipped : type depending on input
+      Decimal values: yield decimal
+      Other numeric values: yield integer (int32)
```

Comment:
I don't think you need to enumerate any types here. Something like "The type of the returned column is the same as type of the input" or something similar.